Editor/Graph/Renderer/DefaultGraphRenderer.cs

```diff
@@ -2,13 +2,14 @@
 using System.Collections.Generic;
 using System.Linq;
 using System.Text.RegularExpressions;
-using UnityEngine;
 using UnityEditor;
+using UnityEngine;
 
 namespace GraphVisualizer
 {
     public class DefaultGraphRenderer : IGraphRenderer
     {
+        public event Action<Node> nodeClicked;
```

Comment:
I'm curious, for what kind stuff do you need to know when a node is clicked inside the graph visualizer? :wink:

Reply:
We use it for traversing larger graphs that don't fit well on a small screen, so it's a callback for specific nodes that we have as "expansion" or "return nodes" (circled here):

![image](https://user-images.githubusercontent.com/904110/52654093-2616fb80-2ea6-11e9-9c5f-952f205502bf.png)

Reply:
I see. I don't think it's the best approach, but the alternative (i.e. zoomable and scrollable canvas) would take way to much time to implement. So if it helps you, it's fine for me right now, until we have a true solution to display more complex graphs.

Does your plan visualizer uses playables as well? it seems your deriving from the PlayableGraph Visualizer to create your visualizer. If it's the case, do you think it would be possible to lower the `nodeClicked` action access (e.g. protected)?

Reply:
API wise, I find it risky to give plain access to a Node to whomever is plugged to this action.

Reply:
Agreed. I implemented this for the reason you mention and because `GraphView` already has support for zooming/scrolling. If you ever want to convert this over to `GraphView` at some point, then I can resurrect a branch I have that has it mostly working.

`PlanVisualizer` doesn't inherit from `PlayableGraphVisualizer` because it was different enough that not much could be shared. Also, `nodeClicked` is defined on `DefaultGraphRenderer` and not `Graph` or `PlayableGraphVisualizer`.

I understand why you think it is risky to give public access to a Node. If you don't like this callback existing in DefaultGraphRenderer, then at the very least we will need to make some methods virtual on DefaultGraphRenderer, so that I can keep the majority of DefaultGraphRenderer and override specific functionality to allow this in a derived class.

Reply:
Marked protected in b8c14ac